assets/aws/files/bin/teleport-generate-config

```diff
@@ -3,92 +3,203 @@ if [[ "${DEBUG}" == "true" ]]; then
-  # this is done because there are two different ways of getting the external hostname
-  # depending on how the AMI is being used and we should support both
-  EXTERNAL_HOSTNAME="${!1}"
-  if [[ "${USE_ACM}" == "true" && "${TELEPORT_PROXY_SERVER_LB}" != "" ]]; then
-    # if an alias for the ACM NLB is configured, also add that to the proxy's kubernetes public_addr
-    if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
-      KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_LB}:3026', '${TELEPORT_PROXY_SERVER_NLB_ALIAS}:3026'"
+    # use the value of the variable name which was passed in
+    # this is done because there are two different ways of getting the external hostname
+    # depending on how the AMI is being used and we should support both
+    EXTERNAL_HOSTNAME="${!1}"
+    if [[ "${USE_ACM}" == "true" && "${TELEPORT_PROXY_SERVER_LB}" != "" ]]; then
+        # if an alias for the ACM NLB is configured, use that for proxy's kubernetes public_addr
+        if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
+            KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_NLB_ALIAS}:3026'"
+        else
+            KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_LB}:3026'"
+        fi
     else
-      KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_LB}:3026'"
+        KUBERNETES_PUBLIC_ADDR="'${EXTERNAL_HOSTNAME:-$PUBLIC_IP}:3026'"
     fi
-  else
-    KUBERNETES_PUBLIC_ADDR="'${EXTERNAL_HOSTNAME:-$PUBLIC_IP}:3026'"
-  fi
-  # set up the kubernetes listener
-  cat << EOS >> /etc/teleport.yaml
+    # set up the kubernetes listener
+    cat << EOS >> ${USE_CONFIG_PATH}
   kubernetes:
     enabled: yes
     listen_addr: 0.0.0.0:3026
     public_addr: [${KUBERNETES_PUBLIC_ADDR}]
 EOS
 }
 
-write_https_keypairs_section() {
-  cat << EOS >> /etc/teleport.yaml
-  https_keypairs:
-    - cert_file: /var/lib/teleport/fullchain.pem
-      key_file: /var/lib/teleport/privkey.pem
+write_ssh_and_tunnel_section() {
+    PASSED_EXTERNAL_TUNNEL_PORT="$1"
+    SSH_PORT=3023
+    if [[ "${TELEPORT_PROXY_SERVER_LB}" != "" ]]; then
+        # ACM
+        if [[ "${USE_ACM}" == "true" ]]; then
+            # if an alias for the ACM NLB is configured, use that for the proxy's ssh_public_addr
+            if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
+                SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
+            else
+                SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
+            fi
+
+            # when using ACM, we have to set up a dedicated listener and target group as
+            # SSL termination is handled by the ACM ALB and it can't multiplex the tunnel connections
+            TUNNEL_PORT="${TUNNEL_LISTEN_PORT}"
+            if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
+                TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
+            else
+                TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
+            fi
```

Comment:
```suggestion
            TUNNEL_PORT="${TUNNEL_LISTEN_PORT}"
            SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
            TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
            
            if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
                # if an alias for the ACM NLB is configured, use that for the proxy's ssh_public_addr
                SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
                # when using ACM, we have to set up a dedicated listener and target group as
                # SSL termination is handled by the ACM ALB and it can't multiplex the tunnel connections
                TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
            fi
```